src/common/src/array/data_chunk_iter.rs

```diff
@@ -288,6 +288,26 @@ impl Row {
         }
         HashCode(hasher.finish())
     }
+
+    /// Compute hash value of a row on corresponding indices.
+    pub fn hash_by_indices<H>(&self, hash_indices: &[usize], hash_builder: &H) -> RwResult<HashCode>
+    where
+        H: BuildHasher,
+    {
+        let mut hasher = hash_builder.build_hasher();
+        for idx in hash_indices {
+            let datum = self.0.get(*idx);
+            match datum {
+                Some(datum) => datum.hash(&mut hasher),
+                None => {
+                    return Err(
+                        ErrorCode::InternalError(format!("index {} out of row bound", idx)).into(),
```

Comment:
Is this Error expected to happen? Dist key should always included by pk right.

Reply:
The error is unwrapped in the caller of this function. The hash function is more like a util function and shouldn't panic.